File: backend/auth-service/src/main/java/com/pangeranvalerensco/orchestria/auth_service/service/AuthService.java
package com.pangeranvalerensco.orchestria.auth_service.service;

import com.pangeranvalerensco.orchestria.auth_service.payload.request.LoginRequest;
import com.pangeranvalerensco.orchestria.auth_service.payload.request.RegisterRequest;
import com.pangeranvalerensco.orchestria.auth_service.payload.response.ApiResponse;
import com.pangeranvalerensco.orchestria.auth_service.payload.response.AuthResponse;
import com.pangeranvalerensco.orchestria.auth_service.payload.response.UserResponse;
import com.pangeranvalerensco.orchestria.auth_service.payload.request.*;
import com.pangeranvalerensco.orchestria.auth_service.payload.response.*;
public interface AuthService {
    
    ApiResponse<UserResponse> register(RegisterRequest request);

    ApiResponse<LoginResult> login(LoginRequest request, String userAgent, String ipAddress, String trustedDeviceToken);
    
    ApiResponse<AuthResponse> verifyOtp(OtpVerifyRequest request, String userAgent, String ipAddress);
    
    ApiResponse<OtpResendResponse> resendOtp(OtpResendRequest request);
    
    ApiResponse<ForgotPasswordStartResponse> forgotPassword(ForgotPasswordRequest request);
    
    ApiResponse<ForgotPasswordVerifyResponse> verifyForgotPassword(ForgotPasswordVerifyRequest request);
    
    ApiResponse<String> resetPassword(PasswordResetRequest request);
    
    ApiResponse<SecuritySettings> getSecuritySettings(String email);
    
    ApiResponse<String> requestEnableTwoFactor(String email);
    
    ApiResponse<String> confirmEnableTwoFactor(String email, OtpVerifyRequest request);
    
    ApiResponse<String> requestDisableTwoFactor(String email);
    
    ApiResponse<String> confirmDisableTwoFactor(String email, OtpVerifyRequest request);

    ApiResponse<UserResponse> getCurrentUser(String email);
    
    String createTrustedDevice(Long userId, String deviceName, String userAgent, String ipAddress);
    
    ApiResponse<java.util.List<TrustedDeviceResponse>> getTrustedDevices(String email);
    
    ApiResponse<String> revokeTrustedDevice(String email, String id);
    
    ApiResponse<String> revokeAllTrustedDevices(String email);
    
    ApiResponse<String> logout(String email, String trustedDeviceToken);
}

File: frontend/src/services/authService.ts
import { apiRequest } from "../api/http";
import type {
  AuthData,
  LoginPayload,
  LoginResult,
  OtpVerifyPayload,
  OtpResendPayload,
  OtpResendResponse,
  ForgotPasswordPayload,
  ForgotPasswordStartResponse,
  ForgotPasswordVerifyPayload,
  ForgotPasswordVerifyResponse,
  PasswordResetPayload,
  SecuritySettings,
  User,
  TrustedDeviceResponse,
} from "../types/auth";

export function loginRequest(payload: LoginPayload) {
  return apiRequest<LoginResult>("/api/auth/login", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function verifyOtp(payload: OtpVerifyPayload) {
  return apiRequest<AuthData>("/api/auth/otp/verify", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function resendOtp(payload: OtpResendPayload) {
  return apiRequest<OtpResendResponse>("/api/auth/otp/resend", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function forgotPassword(payload: ForgotPasswordPayload) {
  return apiRequest<ForgotPasswordStartResponse>("/api/auth/password/forgot", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function verifyForgotPassword(payload: ForgotPasswordVerifyPayload) {
  return apiRequest<ForgotPasswordVerifyResponse>("/api/auth/password/forgot/verify", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function resetPassword(payload: PasswordResetPayload) {
  return apiRequest<string>("/api/auth/password/reset", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function getCurrentUser(token: string) {
  return apiRequest<User>(
    "/api/auth/me",
    {
      method: "GET",
    },
    token,
  );
}

export function getSecuritySettings(token: string) {
  return apiRequest<SecuritySettings>("/api/auth/security", { method: "GET" }, token);
}

export function requestEnableTwoFactor(token: string) {
  return apiRequest<string>("/api/auth/2fa/enable/request", { method: "POST" }, token);
}

export function confirmEnableTwoFactor(token: string, payload: OtpVerifyPayload) {
  return apiRequest<string>("/api/auth/2fa/enable/confirm", {
    method: "POST",
    body: JSON.stringify(payload),
  }, token);
}

export function requestDisableTwoFactor(token: string) {
  return apiRequest<string>("/api/auth/2fa/disable/request", { method: "POST" }, token);
}

export function confirmDisableTwoFactor(token: string, payload: OtpVerifyPayload) {
  return apiRequest<string>("/api/auth/2fa/disable/confirm", {
    method: "POST",
    body: JSON.stringify(payload),
  }, token);
}

export function getTrustedDevices(token: string) {
  return apiRequest<TrustedDeviceResponse[]>("/api/auth/trusted-devices", { method: "GET" }, token);
}

export function revokeTrustedDevice(token: string, id: string) {
  return apiRequest<string>(`/api/auth/trusted-devices/${id}`, { method: "DELETE" }, token);
}

export function revokeAllTrustedDevices(token: string) {
  return apiRequest<string>("/api/auth/trusted-devices", { method: "DELETE" }, token);
}

export function logoutRequest() {
  return apiRequest<string>("/api/auth/logout", { method: "POST" });
}